crates/core/src/text_unparser.rs

```diff
@@ -28,9 +27,9 @@ pub(crate) fn apply_effects<'a>(
     current_name: Option<&str>,
     logs: &mut AnalysisLogs,
 ) -> Result<(String, Option<Vec<Range<usize>>>)> {
-    let mut our_effects = Vec::new();
+    let mut our_effects = Vec::with_capacity(effects.len());
```

Comment:
Tiny optimization :)

Reply:
unpausing is quite slow in files with lots of rewrites. Most files don't have many rewrites, but those that do can be quite slow. We repeatedly clone many strings, I'm sure there are many low hanging fruits there